target-app/Networking/Services/AuthenticationServices.swift

```diff
@@ -35,7 +35,32 @@ class AuthenticationServices {
                     completion(.success(user))
                 } else {
                     //TODO: add error state
-//                    completion(.failure(AuthError.userSessionInvalid))
+                    //completion(.failure(AuthError.userSessionInvalid))
+                }
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    class func signIn(
+        email: String,
+        password: String,
+        completion: @escaping (Result<Void, Error>) -> Void
+    ) {
+        BaseAPIClient.default.request(
+            endpoint: AuthEndpoint.signIn(
+                email: email,
+                password: password
+            )
+        ) { (result: Result<UserResponse?, Error>, responseHeaders: [String: String]) in
+            switch result {
+            case .success(let userResponse):
+                if userResponse != nil {
+                    completion(.success(()))
+                } else {
+                    //TODO: add error state
+                    //completion(.failure(AuthError.userSessionInvalid))
```

Comment:
you need to call the completion closure here or you could leave the caller in an undefined state.

Reply:
done!